kube/src/lib.rs

```diff
@@ -400,4 +420,84 @@ mod test {
         }
         Ok(())
     }
+
+    #[tokio::test]
+    #[ignore] // needs cluster (will create await a pod)
+    #[cfg(all(feature = "runtime"))]
+    async fn pod_can_await_conditions() -> Result<(), Box<dyn std::error::Error>> {
+        use crate::{
+            api::{DeleteParams, PostParams},
+            runtime::wait::{await_condition, conditions, delete::delete_and_finalize, Condition},
+            Api, Client,
+        };
+        use k8s_openapi::api::core::v1::Pod;
+        use std::time::Duration;
+        use tokio::time::timeout;
+
+        let client = Client::try_default().await?;
+        let pods: Api<Pod> = Api::default_namespaced(client);
+
+        // create busybox pod that's alive for at most 30s
+        let p: Pod = serde_json::from_value(serde_json::json!({
+            "apiVersion": "v1",
+            "kind": "Pod",
+            "metadata": { "name": "busybox-kube4" },
+            "spec": {
+                "containers": [{
+                  "name": "busybox",
+                  "image": "busybox:1.34.1",
+                  "command": ["sh", "-c", "sleep 30"],
+                }],
+            }
+        }))?;
+
+        let pp = PostParams::default();
+        match pods.create(&pp, &p).await {
+            Ok(o) => assert_eq!(p.name(), o.name()),
+            Err(crate::Error::Api(ae)) => assert_eq!(ae.code, 409), // if we failed to clean-up
```

Comment:
Wouldn't this still be a fail? Seems like we could easily end up stuck with a false negative that nobody notices otherwise...

Reply:
ah, yes. this a leftover from how the examples would do things - if the thing already exists, keep going in an effort to make it semi-idempotent. it's probably a bit misguided to try to keep it in the tests here now that each pod test has a unique name.

will just change to a `pods.create(...).await?`

Reply:
minor thought occurred; this does make it slightly harder for newcomers to debug tests now if they are doing `ctrl-c` on a test run, and re-running it. i just did this, and managed to hit this again now after changing to `pods.create()?`.

did a follow-up change to add a common label: `app: kube-rs-test` that we then can delete before running integration tests locally (as a precaution - shouldn't be necessary in 99% - but feel it's nice to have it set explicitly somewhere)